msteams-platform/m365-apps/specify-runtime-requirements.md

```diff
@@ -0,0 +1,132 @@
+
+## Microsoft 365 host support
+
+Currently, a subset of Microsoft 365 host applications support the ability to specify runtime requirements in the app manifest. This support will expand over time. The following hosts will ensure only applicable apps and their applicable components are made available to end-users:
+
+|Microsoft 365 host application| Web | Desktop | Mobile | Notes|
+|---|---|---|---|---|
+| Microsoft Teams| ✔️| ✔️| ✔️| |
+| Microsoft Outlook| | | ✔️| |
+
+## Specify relationships between components of your app (`elementRelationshipSet`)
+
+You can specify relationships among the individual components of your app by including a [`elementRelationshipSet`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipset) in your app manifest. Use this object to specify both [one-way dependencies](#one-way-dependencies) and [mutual dependencies](#mutual-dependencies) among app components.
+
+> [!NOTE]
+> A given pair of app components can only be grouped once, by either a `oneWayDependency` or `mutualDependency`.  Similarly, cyclical `oneWayDependencies` (where for example, components *A* and *B* mutually declare each other as one-way dependency) are not allowed and will not pass manifest validation.
+
+Currently, a subset of Teams app elements can be specified as runtime requirements in the app manifest. This support will expand over time. The following app manifest elements can be specified as having one-way or mutual dependencies:
+
+- Personal tabs (`staticTabs`)
+- Message extensions (`composeExtensions`), including individual commands
+- Bots (`bots`)
+- Configurable tabs (`configurableTabs`)
+
+Each app element is denoted by an `id`, which maps to `botId` for bots, `entityId` for staticTabs, `id` for configurableTabs, and `id` for composeExtensions.
+
+> [!NOTE]
+> This feature introduces `id` properties to configurableTabs and composeExtensions definitions. Currently, Teams and other Microsoft 365 hosts only support apps containing a single configurableTab and/or single composeExtension. These ID properties were added to future-proof your app in case host support expands to accommodate multiple instances of these components.
+>
+> The `id` property must be specified on a configurableTab or composeExtension for it to be recognized in an `elementRelationshipSet`.
+
+### One-way dependencies
+
+Use the [`oneWayDependencies`](../resources/schema/manifest-schema-dev-preview.md#elementrelationshipsetonewaydependency) array to describe cases where one component of your app depends upon another component. For each object in the array, specify the dependent component (`element`) and the component it depends on (`dependsOn`). For example:
```

Comment:
> For example:
@erikadoyle, is it necessary to replace this sentence to describe the code snippet briefly (This extends to all the other code snippets in the article)? If you think it isn't necessary, resolve the suggestion.

Suggestion: The following JSON payload shows a message extension with a one-way dependency on a bot: